pkg/volume/util/util.go

```diff
@@ -73,14 +73,28 @@ func UnmountPath(mountPath string, mounter mount.Interface) error {
 		return nil
 	}
 
-	err := mounter.Unmount(mountPath)
-	if err == nil {
-		// Only delete directory on successful unmount
-		glog.V(5).Infof("Unmounted %q. Deleting path.", mountPath)
+	notMnt, err := mounter.IsLikelyNotMountPoint(mountPath)
+	if err != nil {
+		return err
+	}
+	if notMnt {
+		glog.V(4).Info("%q is not a mountpoint, deleting", mountPath)
```

Comment:
On Fri, Aug 19, 2016 at 4:16 PM, Saad Ali <notifications@github.com> wrote:

> In pkg/volume/util/util.go
> <https://github.com/kubernetes/kubernetes/pull/31034#discussion_r75564139>
> :
>
> > @@ -73,14 +73,28 @@ func UnmountPath(mountPath string, mounter mount.Interface) error {
> >  		return nil
> >  	}
> >
> > -	err := mounter.Unmount(mountPath)
> > -	if err == nil {
> > -		// Only delete directory on successful unmount
> > -		glog.V(5).Infof("Unmounted %q. Deleting path.", mountPath)
> > +	notMnt, err := mounter.IsLikelyNotMountPoint(mountPath)
> > +	if err != nil {
> > +		return err
> > +	}
> > +	if notMnt {
> > +		glog.V(4).Info("%q is not a mountpoint, deleting", mountPath)
>
> nit: I would make this a Warningf instead of a low-pri Info.
>
done

> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/31034/files/4a61db70f776c2c61b216012eabeb889bf74c102#r75564139>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ASSNxXx4aIwrgniVXbcDqi0MQy5zrHlXks5qhjligaJpZM4Jo55i>
> .
>



-- 
- Jing